pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -305,6 +305,11 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 			return detailedErr
 		}
 
+		// Successful attach event is useful for user debugging
+		simpleMsg, _ := volumeToAttach.GenerateMsg("AttachVolume.Attach succeeded", "")
+		for _, pod := range volumeToAttach.ScheduledPods {
+			og.recorder.Eventf(pod, v1.EventTypeNormal, kevents.SuccessfulAttachVolume, simpleMsg)
```

Comment:
@saad-ali - maybe we should wait until the new event [design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/events-redesign.md) is out ?

Reply:
Looks like the design proposal is trying to improve event performance as well as improve dev insight into the app.

I believe this event should not impact performance significantly as it is a one-time success event on volume attach (which does not happen often or repeatedly). 

Additionally it gives developers greater insight into debugging their app, as they can now tell in various volume failure modes whether the volume has been attached or not. Specifically: if a volume is successfully attached but a mount call was never made, without this event a developer would have no way of knowing whether an attach call was made, or if the volume is attached anywhere.

Reply:
@davidz627 - the design proposal will introduce a new `#eventf()` that will handle event series. we'll have to migrate to it.

but I guess we can add it for now, and migrate it later.

Reply:
Agreed, I wouldn't block on this. We'll have to migrate the existing code anyway.